cadence/contracts/LendingInterfaces.cdc

```diff
@@ -7,145 +7,145 @@
-        pub fun borrow(userCertificateCap: Capability<&{LendingInterfaces.IdentityCertificate}>, borrowAmount: UFix64): @FungibleToken.Vault
-        pub fun repayBorrow(borrower: Address, repayUnderlyingVault: @FungibleToken.Vault): @FungibleToken.Vault?
-        pub fun liquidate(liquidator: Address, borrower: Address, poolCollateralizedToSeize: Address, repayUnderlyingVault: @FungibleToken.Vault): @FungibleToken.Vault?
-        pub fun flashloan(executorCap: Capability<&{LendingInterfaces.FlashLoanExecutor}>, requestedAmount: UFix64, params: {String: AnyStruct}) {return}
+        access(all) fun supply(supplierAddr: Address, inUnderlyingVault: @{FungibleToken.Vault})
+        access(all) fun redeem(userCertificate: &{LendingInterfaces.IdentityCertificate}, numLpTokenToRedeem: UFix64): @{FungibleToken.Vault}
+        access(all) fun redeemUnderlying(userCertificate: &{LendingInterfaces.IdentityCertificate}, numUnderlyingToRedeem: UFix64): @{FungibleToken.Vault}
+        access(all) fun borrow(userCertificate: &{LendingInterfaces.IdentityCertificate}, borrowAmount: UFix64): @{FungibleToken.Vault}
+        access(all) fun repayBorrow(borrower: Address, repayUnderlyingVault: @{FungibleToken.Vault}): @{FungibleToken.Vault}?
+        access(all) fun liquidate(liquidator: Address, borrower: Address, poolCollateralizedToSeize: Address, repayUnderlyingVault: @{FungibleToken.Vault}): @{FungibleToken.Vault}?
+        access(all) fun flashloan(executor: &{LendingInterfaces.FlashLoanExecutor}, requestedAmount: UFix64, params: {String: AnyStruct}) {return}
     }
 
-    pub resource interface PoolAdminPublic {
-        pub fun setInterestRateModel(newInterestRateModelAddress: Address)
-        pub fun setReserveFactor(newReserveFactor: UFix64)
-        pub fun setPoolSeizeShare(newPoolSeizeShare: UFix64)
-        pub fun setComptroller(newComptrollerAddress: Address)
+    access(all) resource interface PoolAdminPublic {
+        access(all) fun setInterestRateModel(newInterestRateModelAddress: Address)
+        access(all) fun setReserveFactor(newReserveFactor: UFix64)
+        access(all) fun setPoolSeizeShare(newPoolSeizeShare: UFix64)
+        access(all) fun setComptroller(newComptrollerAddress: Address)
         /// A pool can only be initialized once
-        pub fun initializePool(reserveFactor: UFix64, poolSeizeShare: UFix64, interestRateModelAddress: Address)
-        pub fun withdrawReserves(reduceAmount: UFix64): @FungibleToken.Vault
-        pub fun setFlashloanRateBps(rateBps: UInt64)
-        pub fun setFlashloanOpen(isOpen: Bool)
+        access(all) fun initializePool(reserveFactor: UFix64, poolSeizeShare: UFix64, interestRateModelAddress: Address)
+        access(all) fun withdrawReserves(reduceAmount: UFix64): @{FungibleToken.Vault}
+        access(all) fun setFlashloanRateBps(rateBps: UInt64)
+        access(all) fun setFlashloanOpen(isOpen: Bool)
```

Comment:
This interface name makes me think this will be a publicly accessible, but the methods seem to be privileged. Is this interface made public? If so, how is access to these sensitive methods access controlled?

Reply:
Yes these methods are all priviledged ones and aren't exposed through public capabilities. Each LendingPool has its own PoolAdmin resource, but right now all the admin resources are moved to one single admin account.

So the only purpose of this public admin interface is to do batch admin operations in 1 tx like:

```
import FungibleToken from 0xFungibleToken
import LendingInterfaces from 0xLendingInterfaces

transaction(to: Address) {
    prepare(admin: auth(BorrowValue) &Account) {
        // Flow Pool
        var poolAdmin = admin.storage.borrow<&{LendingInterfaces.PoolAdminPublic}>(from: /storage/incrementLendingPoolAdmin_Flow)
            ?? panic("cannot borrow reference to Flow pool admin")
        poolAdmin.setFlashloanRateBps(rateBps: 30)

        // FUSD Pool
        poolAdmin = admin.storage.borrow<&{LendingInterfaces.PoolAdminPublic}>(from: /storage/incrementLendingPoolAdmin_Fusd)
            ?? panic("cannot borrow reference to Fusd pool admin")
        poolAdmin.setFlashloanRateBps(rateBps: 30)

        // USDC Pool
        poolAdmin = admin.storage.borrow<&{LendingInterfaces.PoolAdminPublic}>(from: /storage/incrementLendingPoolAdmin_Usdc)
            ?? panic("cannot borrow reference to Usdc pool admin")
        poolAdmin.setFlashloanRateBps(rateBps: 30)

        // Blt Pool
        poolAdmin = admin.storage.borrow<&{LendingInterfaces.PoolAdminPublic}>(from: /storage/incrementLendingPoolAdmin_Blt)
            ?? panic("cannot borrow reference to Blt pool admin")
        poolAdmin.setFlashloanRateBps(rateBps: 30)

        // stflow Pool
        poolAdmin = admin.storage.borrow<&{LendingInterfaces.PoolAdminPublic}>(from: /storage/incrementLendingPoolAdmin_Stflow)
            ?? panic("cannot borrow reference to Stflow pool admin")
        poolAdmin.setFlashloanRateBps(rateBps: 30)
    }
}
```

Technically we don't need this interface, but it won't bring in security issues and just for the sake of convenience.